src/components/Users/UserProfile.js

```diff
@@ -7,32 +7,70 @@ import Navigation from '../Navigation/Navigation';
 import Footer from '../Footer/Footer';
 import ProfileEvent from './ProfileEvents';
 
-import { userEventsAction } from '../../actions/authEvents.actions';
+import { userEventsAction, deleteEventAction } from '../../actions/authEvents.actions';
+
 
 class UserProfile extends Component {
+
+	constructor(props){
+		super(props);
+		this.state = {
+			toggle:false,
+			selectedId: null,
+			eventDeleted:false
+		};
+		this.toggleModal = this.toggleModal.bind(this);
+		this.onDelete = this.onDelete.bind(this);
+	}
 	componentWillMount(){
-		this.props.dispatch(userEventsAction());
+		this.props.userEventsAction();
 	}
 
-	
+	toggleModal(id) {
+		this.setState({ toggle: !this.state.toggle, selectedId: id});
+	}
+
+	onDelete(id) {
+		let eventId = this.state.selectedId;
+		this.props.deleteEventAction(eventId);
```

Comment:
What happens to the local state if an error occurs during event deletion? you could set state in a `.then()`, that way you would be sure it's only set if the operation was successful.